package proto_ff;

// 这里一定要包含自定义的协议
import "yd_fieldoptions.proto";
import "proto_common.proto";

message Proto_HMLineRateMgr_item
{
  repeated int32 item = 1[(yd_fieldoptions.field_arysize)=15];
}

message Proto_HMLineRateMgr_icon
{
  repeated Proto_HMLineRateMgr_item icon = 1[(yd_fieldoptions.field_arysize)=15];
}

message Proto_HMLineRateMgr_mary
{
  repeated int32 IconRight = 1[(yd_fieldoptions.field_arysize)=15];
}

message Proto_HMLineRateMgr
{
  optional Proto_HMLineRateMgr_icon main = 1;
  optional Proto_HMLineRateMgr_icon free = 2;
  optional Proto_HMLineRateMgr_icon shulaibao = 3;
  optional Proto_HMLineRateMgr_mary mary = 4;
}

message Proto_HMRecordItem_Row
{
  repeated int32 col = 1[(yd_fieldoptions.field_arysize)=5];
}

message Proto_HMRecordItem
{
  repeated Proto_HMRecordItem_Row row = 1[(yd_fieldoptions.field_arysize)=5];
}

message Proto_MaryResult
{
  optional int32 outer_value = 1;
  optional int32 mul = 2;
  repeated int32 vec_inner_value = 3[(yd_fieldoptions.field_arysize)=10];
}

message Proto_LineColCfg_item
{
  repeated int32 item = 1[(yd_fieldoptions.field_arysize)=15];
}

message Proto_LineColCfg_Col
{
  repeated Proto_LineColCfg_item Col = 1[(yd_fieldoptions.field_arysize)=15];
}

message Proto_LineColCfg
{
  optional Proto_LineColCfg_Col  main = 1;  //主游戏列图标配置
  optional Proto_LineColCfg_Col  free = 2;  //免费游戏图标配置
  optional Proto_LineColCfg_Col  shulaibao = 3; //鼠来宝图标配置
  optional Proto_LineColCfg_Col  bubugao = 4; //步步高图标配置
}

message Proto_LinkResult_item
{
  repeated int32 item = 1[(yd_fieldoptions.field_arysize)=10];
}

message Proto_LinkResult_Row
{
  repeated Proto_LineColCfg_item Row = 1[(yd_fieldoptions.field_arysize)=10];
  optional int32 resulttype = 2;
  optional int32 rowcount = 3;
  optional int32 colcount = 4;
}

message Proto_LinkResult
{
  optional Proto_LinkResult_Row config = 1;
}